gcc6 compilation error

When building against gcc 6.2.1 (@archlinux) got this error

```
/tmp/yaourt-tmp-bartus/aur-cmvs-pmvs-git/src/CMVS-PMVS/program/base/cmvs/../stann/dpoint.hpp:492:10: error: no match for ‘operator<<’ (operand types are ‘std::basic_ostream<char>’ and ‘std::istream {aka std::basic_istream<char>}’)
       std::cerr << "Error Reading Point:"
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          << is << std::endl;
          ^~~~~

```

It looks like moving the operator inside the namespace may fix the issue (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=26037#c6)

--- a/program/base/stann/dpoint.hpp
+++ b/program/base/stann/dpoint.hpp
@@ -488,8 +488,9 @@ operator>>(std::istream& is,dpoint<NumType,D> &p)
 	 for (int i=0; i<D; ++i)
 		 if(!(is >> p[i])){
 			 if(!is.eof()){
+			   char errorpoint = is.getloc();
 			   std::cerr << "Error Reading Point:" 
-				     << is << std::endl;
+				     << errorpoint << std::endl;
 				exit(1);
 			 }
 		 }

